prow/pod-utils/clone/clone.go

```diff
@@ -144,10 +170,18 @@ func gitCtxForRefs(refs prowapi.Refs, baseDir string, env []string, oauthToken s
 		g.repositoryURI = refs.CloneURI
 	}
 
-	if len(oauthToken) > 0 {
-		u, _ := url.Parse(g.repositoryURI)
-		u.User = url.UserPassword(oauthToken, "x-oauth-basic")
-		g.repositoryURI = u.String()
+	if token != "" {
+		if user != "" {
+			u, _ := url.Parse(g.repositoryURI)
+			u.User = url.UserPassword(user, token)
+			g.repositoryURI = u.String()
+		} else {
+			// GitHub requires that the personal access token is set as a username.
+			// e.g., https://<token>:x-oauth-basic@github.com/owner/repo.git
+			u, _ := url.Parse(g.repositoryURI)
+			u.User = url.UserPassword(token, "x-oauth-basic")
+			g.repositoryURI = u.String()
```

Comment:
The `u, _ := url.Parse(g.repositoryURI)` line can be before the conditional, and the `g.repositoryURI = u.String()` can be after it. Only the `UserPassword` is conditional.

Reply:
Thank you for your comment. I fixed it in https://github.com/kubernetes/test-infra/pull/23891/commits/390b0c94df6fbba204e772fc4d50648cb8c510f6.